File: src/beakjoon/11729/main.go
package main

import (
	"fmt"
)

type result struct {
	src int
	dest int
}

var output []result

func main() {
	var n int
	
	fmt.Scanf("%d", &n)
	
	hanoi(1, 3, 2, n)
	
	fmt.Println(len(output))
	for _, out := range output {
		fmt.Printf("%d %d\n", out.src, out.dest)
	}
}

func hanoi(src, dest, tmp int, num int) {
	if (num < 1) {
		return
	}
	
	hanoi(src, tmp, dest, num - 1)
	
	output = append(output, result{ src, dest })
	
	hanoi(tmp, dest, src, num - 1)
}

File: src/beakjoon/11729/main.cpp
#include <bits/stdc++.h>

using namespace std;

vector<pair<int, int>> v;
int _cnt;

void hanoi(int src, int dest, int tmp, int n) {
	if (n < 1) {
		return;
	}
	
	hanoi(src, tmp, dest, n - 1);
	
	v.push_back({src, dest});
	
	hanoi(tmp, dest, src, n - 1);
	
	_cnt++;
}

int main(void) {
	int n;
	
	scanf("%d", &n);
	hanoi(1, 3, 2, n);
	
	printf("%d\n", _cnt);
	for (auto& p : v) {
		printf("%d %d\n", p.first, p.second);
	}
	
	return 0;
}
